design/ErrorHandling.md

```diff
@@ -96,61 +96,38 @@ The effect of an error depends on the type of a call:
 
 In each case, an error is logged to the error log.
 
-## *Recovery*: Recoverable Errors
+## *Fallible Allocation*: Out-of-memory in object creation
 
-Recoverable errors are produced only by object creation.
-The status of an object can be retrieved asynchronously (see next section).
-
-```
-enum GPUObjectStatus {
-    // The object is valid.
-    "valid",
-    // The object is invalid due to a non-fatal allocation failure.
-    // The application can use this as a signal to scale back resource usage, if possible.
-    "out-of-memory",
-    // The object is invalid for another, unrecoverable reason.
-    "invalid",
-};
-```
-
-(Note that object creation failures always send an error to the error log, regardless of the object type or the type of failure.)
-
-If an application uses recoverable allocation, the implementation will still generate error log entries:
-a `"recoverable-out-of-memory"` error for the object creation, and `"validation-error"`s for any subsequent uses of the invalid object.
-The application may need to understand whether such error log entries were part of a recovered allocation (e.g. to avoid sending telemetry for those errors).
-To facilitate this filtering, the handle to the invalid object (including "expando" JavaScript properties) is attached to the error log entry (see above).
-
-### Recoverable errors in object creation
-
-A recoverable error is exposed as a `GPUObjectStatusQuery`.
-
-```
-// (Exact form/type subject to change.)
-typedef Promise<GPUObjectStatus> GPUObjectStatusQuery;
-
-typedef (GPUBuffer or GPUTexture) StatusableObject;
+Recoverable fallible allocations are exposed as Promise versions of the `createBuffer`/`createTexture` entry points.
 
+```webidl
 partial interface GPUDevice {
-    GPUObjectStatusQuery getObjectStatus(StatusableObject object);
+    Promise<GPUBuffer?> tryCreateBuffer(GPUBufferDescriptor descriptor);
```

Comment:
Why is it nullable? Shouldn't the API just reject the promise?

Reply:
I talked about this in a [my last comment, just above](https://github.com/gpuweb/gpuweb/pull/197#issuecomment-460776179).

It could reject, but I currently have strong preference to distinguish it via returning null.